modules/Bio/EnsEMBL/Compara/Production/DnaFragChunk.pm

```diff
@@ -312,22 +273,21 @@ sub dump_chunks_to_fasta_file
 
 =head2 dump_loc_file
 
-  Example     : $chunk->dump_loc_file();
-  Description : Returns the path to this Chunk in the dump location of its DnaCollection
+  Arg [1]     : (Optional) string - base directory path. By default, the current
+                working directory.
+  Example     : $chunk->dump_loc_file('/tmp');
+  Description : Returns a unique filepath for this DnaFragChunk.
   Returntype  : String
-  Exceptions  : none
-  Caller      : general
-  Status      : Stable
 
 =cut
 
 sub dump_loc_file {
     my $self = shift;
-    my $dna_collection = shift;
+    my $basedir = shift // cwd();
 
-    my $dump_loc = $dna_collection->dump_loc;
     my $sub_dir  = dir_revhash($self->dbID);
-    return sprintf('%s/%s/chunk_%s.fa', $dump_loc, $sub_dir, $self->dbID);
+    my $filename = sprintf('chunk_%s.fa', $self->dbID);
+    return File::Spec->catfile($basedir, $sub_dir, $filename);
```

Comment:
I left it to avoid saturating the `inode` with too many files. But if you think it is unlikely to generate as many as to affect the performance, I agree it can be removed.